### PR TITLE
#184 #192 new presentation of the results

### PR DESCRIPTION
**Pull request by djakk**
Issues #184 and  #192 : new presentation of the results : mini-calendar, maps, help if no results

capture d ecran 2018-02-13 a 16 32 16
capture d ecran 2018-02-13 a 16 40 29

### CODE REVIEW

**Comment by nicolasleger** [comment]
Please fix conflicts of `config/locales/en.yml`

File: app/controllers/search_controller.rb [hunk 1]
```diff
@@ -7,41 +8,83 @@ def index
 
     @trips ||= []
     @search = Search.new(params[:search])
+    the_number_of_trips_per_page = 10
```
Comment (nicolasleger) [inline app/controllers/search_controller.rb]: Might put it in a constant

File: app/controllers/search_controller.rb [hunk 2]
```diff
@@ -7,41 +8,83 @@ def index
...
+    the_first_date = [(@search.date_value) -31, (Date.today) -31].min # we need trips of the past month to say "hey there was some trips last month" if no results
```
Comment (nicolasleger) [inline app/controllers/search_controller.rb]: Add a constant for around delta of `31`

Comment (nicolasleger) [reply app/controllers/search_controller.rb]: Parenthesis unnecessary

File: app/controllers/search_controller.rb [hunk 3]
```diff
@@ -7,41 +8,83 @@ def index
...
+    the_last_date =  (@search.date_value) +31
```
Comment (nicolasleger) [inline app/controllers/search_controller.rb]: Parenthesis unnecessary

File: app/controllers/search_controller.rb [hunk 4]
```diff
@@ -7,41 +8,83 @@ def index
...
                    .includes(:points)
-                   .where('departure_date >= ?', @search.date_value)
+                   .where('departure_date >= ?', the_first_date)
+                   .where('departure_date <= ?', the_last_date)
                    .where(state: 'confirmed')
                    .from_to(@search.from_lon, @search.from_lat, @search.to_lon, @search.to_lat)
                    .order(departure_date: :asc)
                    .order(departure_time: :asc)
-                   .page(params[:page]).per(10)
+                   #.page(params[:page]).per(the_number_of_trips_per_page)
```
Comment (nicolasleger) [inline app/controllers/search_controller.rb]: no pagination for long list (like from Paris)?

File: app/controllers/search_controller.rb [hunk 5]
```diff
@@ -7,41 +8,83 @@ def index
...
+                   .where('departure_date <= ?', the_last_date)
                    .where(state: 'confirmed')
                    .to_only(@search.to_lon, @search.to_lat)
                    .order(departure_date: :asc)
                    .order(departure_time: :asc)
-                   .page(params[:page]).per(10)
+                   #.page(params[:page]).per(the_number_of_trips_per_page)
       end
     end
+    # set .the_previous_trip__departure_date attribute
```
Comment (nicolasleger) [inline app/controllers/search_controller.rb]: Code below is too fat for error probe check and understanding
You can use private submethod or create a dedicated service / Search

File: app/views/trips/_trip.html.haml [hunk 6]
```diff
@@ -1,17 +1,14 @@
...
         .name= trip.name
         .age= "#{trip.age} ans" if trip.age.present?
     .trip-info
       .title
-        =l trip.departure_date, format: :trip_date
         =l trip.departure_time, format: :short
-        %p= "(le covoit' est déjà parti !)" if (trip.is_before_today?)
-        %p= "/!\\ départ dans #{Integer(trip.departure_date - @search.date_value)} jour(s)" if (trip.is_strictly_before(@search.date_value))
-        %p= "/!\\ départ dans +#{Integer(trip.departure_date - @search.date_value)} jour(s)" if (trip.is_strictly_after(@search.date_value))
+        %p= "(le covoit' est sur la route)" if (trip.is_before_today?)
```
Comment (nicolasleger) [inline app/views/trips/_trip.html.haml]: Parenthesis unnecessary

File: app/controllers/search_controller.rb (hunk 5 above)
Comment (nicolasleger) [reply app/controllers/search_controller.rb]: Poke me if you need help

File: app/controllers/search_controller.rb (hunk 4 above)
Comment (djakk) [reply app/controllers/search_controller.rb]: @nicolasleger well, the pagination is done on the array, the user will still see 10 trips per page (the_number_of_trips_per_page = 10). But that means that there will be plenty of Trip objects created if the request return plenty of rows (the "Paris" case), is that acceptable ?

File: app/controllers/search_controller.rb (hunk 3 above)
Comment (djakk) [reply app/controllers/search_controller.rb]: Are you sure ? .date_value is a function, ```@search.date_value -31``` crashes for me.

**Comment by nicolasleger** [comment]
@djakk Can I resolve conflict on your branch with develop?

**Comment by djakk** [comment]
@nicolasleger sure ! Do I need to authorize you or something ?

**Comment by nicolasleger** [comment]
Conflicts resolved and merged on `develop` which is auto deployed on test site.

**Comment by nicolasleger** [comment]
@djakk Hi, I spotted a small glitch after selecting one date from the input links contains double space like `date%5D=jeudi++8+mars+2018` instead of `date%5D=jeudi+8+mars+2018`.

screen shot 2018-02-18 at 19 57 33

**Comment by nicolasleger** [comment]
You also need that I give you the from around and to around distance filter used (by default `25km` but `1/6th` if 2 cities departure and arrival provided).

**Pull request merged**